mullvad-cli/src/cmds/patch.rs

```diff
@@ -26,10 +26,28 @@ pub async fn handle(source: String) -> Result<()> {
     Ok(())
 }
 
+/// If source is specified, write a patch to the file. Otherwise, write the patch to standard
+/// output.
+pub async fn export(dest: String) -> Result<()> {
+    let mut rpc = MullvadProxyClient::new().await?;
+    let json_blob = rpc
+        .export_json_settings()
+        .await
+        .context("Error exporting patch")?;
+
+    tokio::task::spawn_blocking(|| put_blob(dest, json_blob))
+        .await
+        .unwrap()?;
+
+    Ok(())
+}
```

Comment:
What does `tokio::task::spawn_blocking` do here? Why not just block?

Reply:
It makes no practical difference here because there are no concurrent tasks, but using blocking I/O on tokio worker threads prevents the runtime from making progress on/completing any other futures. So I avoid it in general.

As for why I'm using `std::io` instead of `tokio::io`, that's just because I read that the latter is not recommended for "interactive uses": https://docs.rs/tokio/latest/tokio/io/struct.Stdin.html

Reply:
Can there really be other concurrent futures in the CLI crate? The text we export is also very small, I doubt blocking for that short duration could cause issues.

Reply:
> Can there really be other concurrent futures in the CLI crate?

There are none, but there _could_ be. There probably never will be any.

> The text we export is also very small, I doubt blocking for that short duration could cause issues.

It's not an issue in practice. I just consider it bad in principle to perform (long-running) blocking work on tokio workers.

What are the downsides of doing it this way?

Reply:
Discussed AFK.